addons/payment_razorpay_oauth/models/payment_provider.py

```diff
@@ -131,6 +131,18 @@ def action_razorpay_create_webhook(self):
             },
         }
 
+    # === CONSTRAINT METHODS === #
+
+    @api.constrains('state')
+    def _check_razorpay_credentials(self):
```

Comment:
To make the check's job unique and precise.
```suggestion
    def _check_razorpay_credentials_are_set_before_enabling(self):
```

Reply:
Took it as suggested as perfectly explicit